### DISCUSSION TITLE
When I try https://github.com/prisma/prisma-examples/tree/latest/typescript/graphql-express in serverless env, how to solve below problem as node_modules's folder readonly? only /tmp can read write.

### DISCUSSION THREAD

**Discussion by 45592858** [prisma/prisma#4203] [Q&A] [answered]
`{"errorCode":-1,"errorMessage":"user code exception caught","stackTrace":"Error: EROFS: read-only file system, mkdir '/var/user/node_modules/@types/typegen-nexus-plugin-prisma'\n    at Object.mkdirSync (fs.js:752:3)\n    at Object.createBrandNewDirectorySync [as createSync] (/var/user/node_modules/fs-jetpack/lib/dir.js:62:8)\n    at writeFileSync (/var/user/node_modules/fs-jetpack/lib/write.js:50:11)\n    at Object.writeSync [as sync] (/var/user/node_modules/fs-jetpack/lib/write.js:75:3)\n    at Object.write (/var/user/node_modules/fs-jetpack/lib/jetpack.js:241:13)\n    at Object.exports.hardWriteFileSync (/var/user/node_modules/nexus-plugin-prisma/dist/utils.js:62:8)\n    at doGenerate (/var/user/node_modules/nexus-plugin-prisma/dist/typegen/render.js:50:17)\n    at Object.generateSync (/var/user/node_modules/nexus-plugin-prisma/dist/typegen/render.js:30:5)\n    at new SchemaBuilder (/var/user/node_modules/nexus-plugin-prisma/dist/builder.js:151:21)\n    at Object.build (/var/user/node_modules/nexus-plugin-prisma/dist/builder.js:76:21)"}`

**Comment by ryands17** [comment]
You need to set the `NODE_ENV` environment variable to `production` when running the app so you won't run across this issue. Check [this](https://github.com/ryands17/serverless-prisma/) repo for a whole example on deploying to Lambda.

**Comment by 45592858** [comment]
Thanks for reply. After change the path, seems no error on this:
```
export const schema = makeSchema({
  types: [Query, Mutation, Post, User],
  plugins: [nexusPrisma({
    experimentalCRUD: true,
    outputs: {
        typegen: '/tmp/generated/nexus-prisma.generated.ts'
    }})],
  outputs: {
    schema: __dirname + '/../schema.graphql',
    typegen: '/tmp/generated/nexus.ts'
  },
  typegenAutoConfig: {
    contextType: 'Context.Context',
    sources: [
      {
        source: '@prisma/client',
        alias: 'prisma',
      },
      {
        source: require.resolve('./context'),
        alias: 'Context',
      },
    ],
  },
})
```

But, same problem show like issue [https://github.com/prisma/prisma/discussions/4204](url)
```
------------------------
run query 
query {
  feed {
    id
    title
    content
    published
    author {
      id
      name
      email
    }
  }
}
----------------------------
return
{
  "errors": [
    {
      "message": "\nInvalid `prisma.post.findMany()` invocation in\n/var/user/src/schema.js:34:40\n\n\n  EROFS: read-only file system, chmod '/var/user/node_modules/.prisma/client/query-engine-rhel-openssl-1.0.x'",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "feed"
      ]
    }
  ],
  "data": {
    "feed": null
  }
}

**Reply by ryands17** [reply]
Follow this [repo](https://github.com/ryands17/serverless-prisma/) as this is a full working example on deploying a GraphQL app to AWS Lambda. You need to include the binaries, ship the `node_modules` and run the app setting `NODE_ENV` to `production`.

**Reply by 45592858** [reply]
When I tried [repo](https://github.com/ryands17/serverless-prisma/) with postgresql DB, after check out the code and config the DB url, run "yarn" to install dependencies, and run "yarn db:save", show below error:

D:\SevenFiles\MyWorks\SRC\liangtuo.cn\try\serverless-prisma-master>yarn db:save
yarn run v1.22.10
$ yarn prisma migrate save --experimental
$ dotenv -- prisma migrate save --experimental
Environment variables loaded from D:\SevenFiles\MyWorks\SRC\liangtuo.cn\try\serverless-prisma-master\.env
Prisma schema loaded from prisma\schema.prisma
Error: undefined: Error querying the database: Error querying the database: db error: ERROR: prepared statement "s0" already exists


error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

D:\SevenFiles\MyWorks\SRC\liangtuo.cn\try\serverless-prisma-master>

**Reply by ryands17** [reply]
Try setting `provider = "postgresql" in the `schema.prisma` to make sure it is connecting to postgres.

**Reply by 45592858** [reply]
yes, already set to postgresql and the connection is normal.

**Reply by ryands17** [reply]
Are you directly using Postgres or are you using something like PgBouncer?
If you're using PgBouncer, then add `pgbouncer=true` in your database URL connection string.

If you're not using PgBouncer, you can try adding `statement_cache_size=0` to the connection string too prevent the error you're facing.

For e.g.
like this:

```
DB_URL = "postgresql://username:password@pgHost:5432/dbName?statement_cache_size=0"
```

**Reply by 45592858** [reply]
"prepared statement "s0" already exists" problem solved after set "statement_cache_size=0"
"ROFS: read-only file system, chmod '/var/user/node_modules/.prisma/client/query-engine-rhel-openssl-1.0.x'" problem still don't know how to solve after check the code again.

**Reply by 45592858** [reply, marked as answer]
"ROFS: read-only file system" problem solved when I use a jenkeins server(linux) to deploy.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
